iommu_ref_model/libtables/src/translate_gpa.c

```diff
@@ -5,7 +5,7 @@
 #include "iommu.h"
 uint64_t
 translate_gpa (
-    iohgatp_t iohgatp, uint64_t gpa, uint64_t *spa) {
+    iohgatp_t iohgatp, uint8_t gxl, uint64_t gpa, uint64_t *spa) {
```

Comment:
Could we instead pass iommu as input to translate_gpa?  This would make the API stable for any future change and will match the prototype of other functions which take the iommu pointer as the first arguement. 

 ```
uint64_t
 translate_gpa (
     iommu_t *iommu, iohgatp_t iohgatp, uint64_t gpa, uint64_t *spa) {
```

GXL can be obtained inside translate_gpa as follows:

`if ( iohgatp.MODE == IOHGATP_Sv39x4 && iommu->reg_file.fctl.gxl == 0 ) {`


The callers then need to pass the iommu pointer:

```
-        if ( translate_gpa(iohgatp, a, &a) == -1) return -1;
+        if ( translate_gpa(iommu, iohgatp, a, &a) == -1) return -1;

```